### PR TITLE
Fix SSR bugs

### PR DESCRIPTION
**Pull request by mxstbr**
**Status**
- [ ] WIP
- [x] Ready for review
- [ ] Needs testing

**Deploy after merge (delete what needn't be deployed)**
- hyperion (frontend)

**Release notes for users (delete if codebase-only change)**
- Fixed a bunch of misc bugs

**Todo**

- [x] Chat input sometimes merging with "join" upsell
- [x] DM threads list no scrollable
- [ ] ~~Thread subscribed status not updating in real-time~~

### CODE REVIEW

**Comment by mxstbr** [comment]
The thread subscribed status not updating in real-time has nothing to do with SSR. It always works in the inbox view, and it never works on the thread view, no matter whether it's client-side rendered or server-side rendered. That makes sense if you think about it, because what updates the thread data once you're subscribed? The `listenToNewThreads` GraphQL subscription that runs in the inbox view—which does not run on the thread view.

We can fix that by running a subscription on the current threads' channel when looking at the thread view, but let's postpone that fix because it's not a big deal. That leaves the DM threads list not infinite scrolling, will dig into that!

File: src/views/directMessages/components/threadsList.js [hunk 1]
```diff
@@ -80,7 +80,6 @@ class ThreadsList extends React.Component<Props, State> {
           hasMore={hasNextPage}
           loader={<LoadingDM />}
           useWindow={false}
-          initialLoad={false}
```
Comment (mxstbr) [inline src/views/directMessages/components/threadsList.js:83]: @brianlovin this prop `initialLoad={false}` is used throughout the app
in more places—and I assume it breaks infinite scrolling after SSR in
all of those places? Why was that introduced in the first place, what's
it needed for? Removing it fixes the issue and doesn't _seem_ to break
anything, but I'm not sure if I'm just missing something...

Comment (brianlovin) [reply src/views/directMessages/components/threadsList.js:83]: Honestly no clue...let's try removing it and test on alpha for a few days.

**Comment by brianlovin** [comment]
Last thing here would be figuring out why we're seeing 9 day old notifications in the dropdown:

screenshot 2018-06-22 10 57 25

And when we see these stale notifications, the 'mark all as read' button doesn't work

**Comment by mxstbr** [comment]
> Last thing here would be figuring out why we're seeing 9 day old notifications in the dropdown:

Do you happen to have a consistent repro for that by chance? :angel:

Let's just ship this for now to solve these annoying bugs and then tackle the notification one in a separate PR? (I don't even know if that's a SSR thing, and it's not worth blocking this PR for that fix)

**Review by brianlovin — APPROVED** [review]

**Pull request merged**